### DISCUSSION TITLE
Siemens/Unify SIP 404 Not Found on Register

### DISCUSSION THREAD

**Discussion by Evope** [baresip/baresip#1884] [Q&A] [answered]
Aloha :)

I have a short little question.
I try to register my SIP-Account with baresip in Unify OpenScape 4000 (formerly Siemens HiPath 4000).
It is not working because the PBX evaluates the Contact header in Register packet:
`Contact: <sip:519-0x33fa54@192.168.115.138:36441;transport=tcp>;expires=3700;+sip.instance="<urn:uuid:9554feef-aa97-8a3c-71e0-8e3e68a0cb68>"`
The extension in the PBX is 519 and as expected the PBX shows:
`fFindSipEndpointObj():Object 519-0x33fa54 not found`
And resulting:
`SIP/2.0 404 Not Found`

In issues tab I found some entries, the last one was https://github.com/baresip/baresip/issues/1644
Sadly, the workaround that exse2 suggested is not working anymore.

So I may ask you frendly if you have any ideas how I can bend the baresip in that way, that in Contact-Header will only be the extension transmitted.
I know that this is in general a problem in the Unify device. But I can imagine that I get much earlier and better support from you than from Unify. And if it doesn't work without major effort, then I accept that.

Thanks

**Comment by sreimers** [comment, marked as answer]
I had a quick look and the mentioned workaround should normally work in the same way as before. If you press `u` or `/uastat` what does the `cuser` field look like?

```diff
diff --git a/src/ua.c b/src/ua.c
index b5e63a44..693afbec 100644
--- a/src/ua.c
+++ b/src/ua.c
@@ -948,7 +948,7 @@ int ua_alloc(struct ua **uap, const char *aor)
 
        /* generate a unique contact-user, this is needed to route
           incoming requests when using multiple useragents */
-       err = re_sdprintf(&ua->cuser, "%r-%p", &ua->acc->luri.user, ua);
+       err = re_sdprintf(&ua->cuser, "%r", &ua->acc->luri.user);
        if (err)
                goto out;
```

**Reply by Evope** [reply]
I made the changes in ua.c like you did. `cuser` says:
`cuser:     519-0x2036c24`
Just to make this clear, I have to make the changes in `~/baresip/src/ua.c` right?

**Reply by sreimers** [reply]
If `~/baresip` is your baresip repo/source folder, then yes. how do you compile?

**Reply by Evope** [reply]
Well, I just followed the instructions under https://github.com/baresip/baresip/wiki/Install:-Stable-Release.
So actually the directory is `~/baresip-2.4.0/src/` but this shouldn't matter.

What do you mean with compile? I just transferred the ua.c file to my computer, (made a backup), changed the marked line via Notepad++:
`err = re_sdprintf(&ua->cuser, "%r-%p", &ua->acc->luri.user, ua);`
to
`err = re_sdprintf(&ua->cuser, "%r", &ua->acc->luri.user);`
then saved and copied back to the mentioned directory.

After I realised that its not working, I also tried to type `/conf_reload` an did a reboot on the raspberry.

**Reply by Evope** [reply]
I am sorry but I did not know that I have to compile the Software again.
So now I did a `make` and `sudo make install` and it works. `cuser` in `/ustat` shows just the extension and registration works.
Thanks for the Support <3
